src/metabase/models/field_usage.clj

```diff
@@ -0,0 +1,100 @@
+(ns metabase.models.field-usage
+  (:require
+   [metabase.lib.core :as lib]
+   [metabase.lib.dispatch :as lib.dispatch]
+   [metabase.lib.hierarchy :as lib.hierarchy]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.util :as lib.util]
+   [metabase.lib.util.match :as lib.util.match]
+   [metabase.models.interface :as mi]
+   [metabase.query-processor.middleware.fetch-source-query :as fetch-source-query]
+   [metabase.util.malli :as mu]
+   [methodical.core :as methodical]
+   [toucan2.core :as t2]
+   [toucan2.tools.disallow :as t2.disallow]))
+
+(doto :model/FieldUsage
+  (derive :metabase/model)
+  (derive ::t2.disallow/update)
+  (derive :hook/created-at-timestamped?))
+
+(methodical/defmethod t2/table-name :model/FieldUsage [_model] :field_usage)
+
+(t2/deftransforms :model/FieldUsage
+  {:used_in                mi/transform-keyword
+   :aggregation_function   mi/transform-keyword
+   :breakout_temporal_unit mi/transform-keyword
+   :breakout_binning       mi/transform-json
+   :filter_op              mi/transform-keyword
+   :filter_args            mi/transform-json})
+
+(defn- filter->field-usage
+  [query stage filter-clause]
+  (let [filter-parts (lib/filter-parts query stage filter-clause)
+        field-id     (-> filter-parts :column :id)]
+    (when (int? field-id)
+      {:field_id    field-id
+       :used_in     :filter
+       :filter_op   (-> filter-parts :operator :short)
+       :filter_args (:args filter-parts)})))
+
+(defn- aggregation->field-usage
+  [query stage aggregation-clause]
+  (let [aggregation-column (lib/aggregation-column query stage aggregation-clause)
+        field-id           (:id aggregation-column)]
+    (when (int? field-id)
```

Comment:
Is `field-id=nil` being covered in the tests? I might have missed it

Reply:
added an aggregation over a "sum" column in 1dcf8c9a6ce44defc4a9aa95fa903d2ea80fdb1c.

not sure how extensive we want to test this but it looks safe to me.

Reply:
cheers, that works

Reply:
I have a similar question for the other clause types though, not just aggregations. is there coverage for `filter`, `breakout`, and `expression`?

Reply:
added some more dummy clause that will have string field-id in [ea4aff0](https://github.com/metabase/metabase/pull/39671/commits/ea4aff04d5d1a2d1c113db614a5b603841a0d45a)